### PR TITLE
Expose OfflineRegion/OfflinePack pack state in progress object and add resume/suspend offline download support

### PR DESCRIPTION
**Pull request by atarsha**
Android's OfflineRegionStatus differs from iOS's MGLOfflinePackState upstream. Since React Native is cross-platform, I also normalize the output of state in the progress object so that Android and iOS return the same integer for the same state in the progress object.

### CODE REVIEW

**Comment by atarsha** [comment]
@zugaldia If you have an opportunity to review this, I would appreciate the feedback. I'm not a Java developer, normally, so I tried to mimic the existing code as much as possible.

Also, curious your thoughts on how I normalized the state since RN is cross-platform.

**Comment by atarsha** [comment]
If you don't mind, hold off on the review. I'm working on bringing offline region/offline pack resume and suspend to RN also since state, suspend and resume are all closely related.

**Comment by atarsha** [comment]
Okay, feel free to review now. I have also added support for suspending and resuming OfflineRegion/OfflinePack downloads and updated the docs and example.

This includes an update to the native SDK version which resolves the issues in https://github.com/mapbox/react-native-mapbox-gl/issues/525 and https://github.com/mapbox/react-native-mapbox-gl/issues/562

**Comment by atarsha** [comment]
@zugaldia can we get this reviewed and if it looks good, merged? Is there someone else at Mapbox who might be available soon to do so?

**Comment by atarsha** [comment]
@alinapaz Can you help us get eyes on this and a few other PRs? There's a lot of great work in PR on this repo just waiting for somebody who has the keys to merge.

**Comment by nolan-m** [comment]
Thanks for this.  I needed this functionality for a current project.  Works great for resuming offline downloads.

**Comment by zugaldia** [comment]
Hey, I just wanted to give y'all an update - good news, soon the Mapbox team is gonna grow to have dedicated resources for this project 🎉 . This means that we'll be able to move much faster fixing and merging PRs, please bear with us while we internally cross the t's and dot the i's.

**Comment by ddolheguy** [comment]
@zugaldia That is AWESOME news....

**Comment by nitaliano** [comment]
@atarsha I'll look this over mind fixing the conflicts?

**Comment by atarsha** [comment]
@nitaliano Resolved! Thanks!

**Comment by atarsha** [comment]
@nitaliano I know you have a lot on your plate getting caught up here, but is there any chance this will get reviewed anytime soon? I've been waiting for this for two months, and would love to get off a forked repo. Thank you!

**Comment by nitaliano** [comment]
@atarsha I'll get it reviewed tomorrow then we can get it merged in.  I also plan on testing this PR out with https://github.com/mapbox/react-native-mapbox-gl/pull/553 that updates the mapbox android SDK to a new major version. I glanced over the code in here already and I don't think there will be any breaking changes.

File: android/src/main/java/com/mapbox/reactnativemapboxgl/ReactNativeMapboxGLModule.java
Comment (nitaliano) [inline android/src/main/java/com/mapbox/reactnativemapboxgl/ReactNativeMapboxGLModule.java]: I think it would be good if we made some static constants for the state so the code can be self documenting here.

File: android/src/main/java/com/mapbox/reactnativemapboxgl/ReactNativeMapboxGLModule.java
Comment (nitaliano) [inline android/src/main/java/com/mapbox/reactnativemapboxgl/ReactNativeMapboxGLModule.java]: Did you mean for this to be commented out? If not could this be removed

File: android/src/main/java/com/mapbox/reactnativemapboxgl/ReactNativeMapboxGLModule.java
Comment (nitaliano) [inline android/src/main/java/com/mapbox/reactnativemapboxgl/ReactNativeMapboxGLModule.java]: Same here

File: android/src/main/java/com/mapbox/reactnativemapboxgl/ReactNativeMapboxGLModule.java
Comment (nitaliano) [inline android/src/main/java/com/mapbox/reactnativemapboxgl/ReactNativeMapboxGLModule.java]: We could break out finding the observer into it's own method like `private OfflineRegionProgressObserver getObserver(String name)` and we can then initialize the variable as `final OfflineRegionProgressObserver foundObserver` instead of having to initialize `foundObserver` and `_foundObserver`. Will also help with reducing code

**Review by nitaliano — CHANGES_REQUESTED** [review]
Other than a few small nits on the Android side this looks fantastic!

**Comment by atarsha** [comment]
@nitaliano I made the suggested changes, although I am not currently near my normal development machine, so I am not able to test them. If you want to give it a shot before merging, it would be advisable.

**Review by nitaliano — APPROVED** [review]

**Pull request merged**
